### ISSUE TITLE
[Suggestion]: Allow check of block state or NBT when determining container spoil modifier

### ISSUE THREAD

**Issue by arketec** [Mrbysco/Spoiled#60]
### General Info

- [X] I would like to propose new content
- [ ] I would like to propose changes
- [ ] I would like to propose another thing

### Suggestion Description

I would be great to have the ability to define a block state and/or NBT requirement in addition to the container block to modify the spoilage rate.

As an example, I have a powered cooler created with Custom Machinery and I would like the ability to have it modify (reduce) the spoilage rate, but only when it is provided power. This can be done with a block state check or an nbt tag check, either would work but I figure implementing one isn't much less than implementing both

maybe something like

[] for blockstate checks
```
containerModifier = ["mynamespace:myblock[Status=RUNNING],0.05"]
```
{} for nbt checks

```
containerModifier = ["mynamespace:myblock{sometag=powered},0.05"]
```

**Comment by Mrbysco** [comment]
Issue is, the container modifier only accepts block entity ids which are not the same as block ids

**Comment by arketec** [comment]
Yeah, I actually took a stab at implementing it myself by changing the cache config to store an object with the modifier as a field and block state and nbt tag as optional fields, but I ran into issues converting the strings into Tag/Blockstates (I just don’t know of the mechanism to do so)
